### ISSUE TITLE
CRPS loss function

### ISSUE THREAD

**Issue by xnuohz** [dmlc/xgboost#10710]
Can xgb support self-defined loss function like this?  multi-classification problem, can `data.get_label()` support type of list?

```
def calc_crps_loss(self, y: Tensor, pred: Tensor) -> Tensor:
        dist = torch.cumsum(pred, dim=1)
        loss = torch.pow(y - dist, 2)
        idx = torch.where(y == -1)
        loss[idx[0], idx[1]] = 0
        return loss.mean()
```

**Comment by xnuohz** [comment]
Thanks for the reply. In my case, y ==> [B, n_class], pred ==> [B, n_class]
Seems xgb does not support multi-dimensional targets. Could you give an example of a function that handles this format?

**Comment by trivialfis** [comment]
https://xgboost.readthedocs.io/en/latest/tutorials/advanced_custom_obj.html might help.

**Comment by trivialfis** [comment]
Feel free to reopen if there are further questions
